src/v/kafka/server/server.cc

```diff
@@ -780,6 +781,88 @@ offset_fetch_handler::handle(request_context ctx, ss::smp_service_group) {
     co_return co_await ctx.respond(std::move(resp));
 }
 
+template<>
+ss::future<response_ptr>
+offset_delete_handler::handle(request_context ctx, ss::smp_service_group) {
+    offset_delete_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    if (!ctx.authorized(
+          security::acl_operation::remove, request.data.group_id)) {
+        co_return co_await ctx.respond(
+          offset_fetch_response(error_code::group_authorization_failed));
+    }
+
+    /// Remove unauthorized topics from request
+    auto unauthorized_it = std::partition(
+      request.data.topics.begin(),
+      request.data.topics.end(),
+      [&ctx](const offset_delete_request_topic& topic) {
+          return ctx.authorized(security::acl_operation::remove, topic.name);
```

Comment:
why authorize the `remove` operation as opposed to `read` or something else?

Reply:
Its what kafka does:
```
  def handleOffsetDeleteRequest(request: RequestChannel.Request, requestLocal: RequestLocal): Unit = {
    val offsetDeleteRequest = request.body[OffsetDeleteRequest]
    val groupId = offsetDeleteRequest.data.groupId

    if (authHelper.authorize(request.context, DELETE, GROUP, groupId)) {
...
```

Reply:
that's validation for a group operation. the comment is about topic operation.

Reply:
Ah yes it should be `read`